.github/workflows/build.yml

```diff
@@ -0,0 +1,125 @@
+name: Build
+
+on:
+    push:
+        branches: [ master ]
+    pull_request:
+    release:
+        types: [ created ]
+
+jobs:
+    tests:
+        runs-on: ubuntu-latest
+        name: Build and test
+        strategy:
+            fail-fast: false
+            matrix:
+                php: [ "7.3" ]
+                composer-version: [ "1" ]
+                coverage: [ "none" ]
+                publish-phar: [ false ]
+                phar-readonly: [ false ]
+                git-fetch-depth: [ 1 ]
+                dependencies: [ "locked" ]
+                include:
+                    -   php: "7.2"
+                        composer-version: "1"
+                        git-fetch-depth: 1
+                        dependencies: "lowest"
+                    -   php: "7.3"
+                        composer-version: "1"
+                        publish-phar: true
+                        git-fetch-depth: 0 # box/composer needs history to determine a recent git version
+                        dependencies: "locked"
+                    -   php: "7.3"
+                        composer-version: "1"
+                        phar-readonly: true
+                        git-fetch-depth: 1
+                        dependencies: "locked"
+                    -   php: "7.4"
+                        composer-version: "1"
+                        coverage: "pcov"
+                        git-fetch-depth: 1
+                        dependencies: "locked"
+                    -   php: "7.4"
+                        composer-version: "2"
+                        git-fetch-depth: 1
+                        dependencies: "locked"
+
+        steps:
+            -   uses: actions/checkout@v2
+                with:
+                    fetch-depth: "${{ matrix.git-fetch-depth }}"
+
+            -   name: Setup PHP
+                uses: shivammathur/setup-php@v2
+                with:
+                    php-version: "${{ matrix.php }}"
+                    ini-values: "phar.readonly=0"
+                    tools: "composer:v${{ matrix.composer-version }}"
+                    coverage: "${{ matrix.coverage }}"
+
+            -   name: Get composer cache directory
+                id: composercache
+                run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            -   name: Cache composer dependencies
+                uses: actions/cache@v2
+                with:
+                    path: ${{ steps.composercache.outputs.dir }}
+                    key: ${{ runner.os }}-composer-cache-${{ hashFiles('**/composer.lock') }}
+                    restore-keys: ${{ runner.os }}-composer-cache-
+
+            -   name: "Install lowest dependencies"
+                if: matrix.dependencies == 'lowest'
+                run: composer update --prefer-lowest --no-interaction --no-progress --no-suggest --prefer-dist
```

Comment:
This part should be removed since it changes `composer.lock` and makes the tests fail.
It also highlights a (minor) problem in Travis: `--prefer-lowest` has no effect the way it is used